cache/lib/cache_web/controllers/key_value_controller.ex

```diff
@@ -0,0 +1,43 @@
+defmodule CacheWeb.KeyValueController do
+  use CacheWeb, :controller
+  alias Cache.Authentication
+  alias Cache.KeyValueStore
+
+  def get_value(conn, %{"account_handle" => account_handle, "project_handle" => project_handle, "cas_id" => cas_id}) do
+    case Authentication.ensure_project_accessible(conn, account_handle, project_handle) do
```

Comment:
The case `Authentication.ensure_project_accessible` is repeated across functions, what about adhering to the plug contract and define it as a plug at the controller level:

```exs
plug CacheWeb.Authentication
```